### PR TITLE
Added button to use current location

### PR DESCRIPTION
**Pull request by DracoBlue**
It would be great to move the map to the current location of the browser.

## Description
It adds a button to use HTML5 geoapi to retrieve current location and sets the current marker to the position of this map.

## Motivation and Context
It makes it easier to move the map center to the place where you are right now.

## How Has This Been Tested?
Clicked that button with geo activated and the map moved to the right place. If detection is not available a message is shown, as soon as one hovers the button.

## Screenshots (if appropriate):

Before in action:
bildschirmfoto 2016-07-23 um 11 38 02

While working:
bildschirmfoto 2016-07-23 um 11 38 06

## Types of changes
- [x] New feature: Add "Use current location"-Button

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation: No?!
- [x] I have updated the documentation accordingly: No!

### CODE REVIEW

**Comment by MRokas** [comment]
NOTE: Wont work in chrome without https connection

File: static/map.js [hunk 1]
```diff
@@ -158,6 +158,34 @@ function initSidebar() {
     $('#scanned-switch').prop('checked', localStorage.showScanned === 'true');
     $('#sound-switch').prop('checked', localStorage.playSound === 'true');
 
+    var browserLocationButton = $(document.getElementById('browser-location'));
```
Comment (tegob) [inline static/map.js]: Use `$('#browser-location')` instead.

File: static/map.js [hunk 2]
```diff
@@ -158,6 +158,34 @@ function initSidebar() {
...
+            $("#next-location").val("");
+            map.setCenter(loc);
+            marker.setPosition(loc);
+            browserLocationButton.removeAttr('disabled');
+            browserLocationButton.text(browserLocationButtonDefaultText);
+          });
+        });
+      });
+    } else {
+      browserLocationButton.attr('disabled', 'disabled');
```
Comment (tegob) [inline static/map.js]: Better: `browserLocationButton.prop('disabled', true)`

File: static/map.js [hunk 3]
```diff
@@ -158,6 +158,34 @@ function initSidebar() {
...
+      browserLocationButton.attr('title', 'You cannot use geolocation, since your browser does not support this.');
```
Comment (tegob) [inline static/map.js]: Use `.prop` here as well

File: static/map.js [hunk 4]
```diff
@@ -158,6 +158,34 @@ function initSidebar() {
     $('#scanned-switch').prop('checked', localStorage.showScanned === 'true');
     $('#sound-switch').prop('checked', localStorage.playSound === 'true');
 
+    var browserLocationButton = $(document.getElementById('browser-location'));
+    var browserLocationButtonDefaultText = browserLocationButton.text();
+    if ("geolocation" in navigator) {
```
Comment (tegob) [inline static/map.js]: Use single quotes for consistency

File: templates/map.html [hunk 5]
```diff
@@ -51,6 +51,12 @@
 				</label>
 				<hr width="100%">
 			</div>
+			<div class="form-control">
+				<label for="browser-location">
+					<h3>Detect location</h3>
+					<button id="browser-location" name="browser-location" placeholder="Change your location">Use current location</button>
```
Comment (tegob) [inline templates/map.html]: “Change your location” is not a good `placeholder` value. Either give it a better value or remove it altogether.

[The HTML spec](https://html.spec.whatwg.org/multipage/forms.html#the-placeholder-attribute) says:

> The `placeholder` attribute should not be used as an alternative to a `<label>`.

File: static/map.js (hunk 1 above)
Comment (DracoBlue) [inline static/map.js]: done.

File: static/map.js (hunk 2 above)
Comment (DracoBlue) [inline static/map.js]: done.

File: static/map.js (hunk 3 above)
Comment (DracoBlue) [inline static/map.js]: done.

File: static/map.js (hunk 4 above)
Comment (DracoBlue) [inline static/map.js]: done

File: templates/map.html (hunk 5 above)
Comment (DracoBlue) [inline templates/map.html]: done.

**Comment by ngg** [comment]
I would prefer this PR:
https://github.com/AHAAAAAAA/PokemonGo-Map/pull/1755

That supports the same functionality and some more (it's possible to double click anywhere on the map to set the location)

**Comment by coolaj86** [comment]
I've got this implemented in my node fork (https://github.com/Daplie/node-pokemap and https://github.com/Daplie/PokemonGo-Map-web-client), but it's not so pretty (not a css guy).

I have it more up in the user's face and then save the preference in local storage (and use the watch location feature in browsers that support it).

If you'd care to spruce up my button I'll give you write access to merge it.

**Comment by coolaj86** [comment]
I've got this implemented in my node fork (https://github.com/Daplie/node-pokemap and https://github.com/Daplie/PokemonGo-Map-web-client), but it's not so pretty (not a css guy).

I have it more up in the user's face and then save the preference in local storage (and use the watch location feature in browsers that support it).

If you'd care to spruce up my button I'll give you write access to merge it.

**Pull request closed without merging**